lib/commands/xctest.js

```diff
@@ -0,0 +1,209 @@
+import B from 'bluebird';
+import { logger } from 'appium-support';
+import log from '../logger';
+
+let commands = {};
+
+const XCTEST_TIMEOUT = 60 * 60 * 1000;
+
+const xctestLog = logger.getLogger('XCTest');
+
+/**
+ * Asserts that IDB is present and that launchWithIDB was used
+ *
+ * @param {object} opts Opts object from the driver instance
+ */
+export function assertIDB (opts) {
+  if (!opts.device?.idb || !opts.launchWithIDB) {
+    log.errorAndThrow(`To use XCTest runner, IDB (https://github.com/facebook/idb) must be installed ` +
+      `and sessions must be run with the "launchWithIDB" capability`);
+  }
+  return opts.device.idb;
+}
+
+/**
+ * Parse the stdout of XC test log
+ * @param {string} stdout A line of standard out from `idb xctest run ...`
+ */
+export function parseXCTestStdout (stdout) {
+  // Parses a 'key' into JSON format
+  function parseKey (name) {
+    const words = name.split(' ');
+    let out = '';
+    for (const word of words) {
+      out += word.substr(0, 1).toUpperCase() + word.substr(1);
+    }
+    return out.substr(0, 1).toLowerCase() + out.substr(1);
+  }
+
+  // Parses a 'value' into JSON format
+  function parseValue (value) {
+    switch (value.toLowerCase()) {
+      case 'true': return true;
+      case 'false': return false;
+      case '': return null;
+      default: break;
+    }
+    if (!isNaN(value)) {
+      if (value.indexOf('.') > 0) {
+        return parseFloat(value);
+      }
+      return parseInt(value, 10);
+    }
+  }
+  if (stdout) {
+    // Parse each line into an array
+    const lines = stdout.trim().split('\n');
+
+    // One single string, just return the string
+    if (lines.length === 1 && !lines[0].includes('|')) {
+      return lines[0];
+    }
+
+    const results = [];
+    for (const line of lines) {
+      // The properties are split up by pipes and each property
+      // has the format "Some Key : Some Value"
+      const properties = line.split('|');
+
+      // Parse each property
+      const output = {};
+      let entryIndex = 0;
+      for (let prop of properties) {
```

Comment:
consider using entries() for enumeration in loops: https://stackoverflow.com/questions/34336960/what-is-the-es6-equivalent-of-python-enumerate-for-a-sequence